lib/org/chromium/webidl/Canonicalizer.js

```diff
@@ -0,0 +1,139 @@
+
+  documentation: `This component expects to receive AST nodes. Canonical versions
+    of definitions will be created from AST nodes.
+
+    Please note that canonicalization for same IDL fragment repository cannot
+    be distributed. All computations for the same repository must be done
+    through one canonicalizer instances, or you will get fragmented results.`,
+
+  requires: [
+    'org.chromium.webidl.ast.Enum',
+    'org.chromium.webidl.ast.Typedef',
+    'org.chromium.webidl.PipelineMessage',
+  ],
+
+  imports: [ 'error' ],
+
+  properties: [
+    // TODO: This component should not be responsible for observing the
+    // pipeline. onDone and waitTime should be removed in the future and
+    // replaced with a subscription to a component that is responsible for
+    // observing the pipeline.
+    {
+      class: 'Function',
+      documentation: `onDone is called after waitTime seconds has passed since
+        the last call to addFragment. We currently have no way of determining
+        when the Canonicalizer is finished receiving fragments. It is assumed
+        that if waitTime has passed and no new fragments are received, then
+        it will not receive any more fragments and results can be passed on.`,
+      name: 'onDone',
+      required: true,
+    },
+    {
+      class: 'Int',
+      documentation: `The amount of time to wait in seconds after the last
+        addFragment request, before onDone() is called with the canonical
+        IDL files.`,
+      name: 'waitTime',
+    },
+    {
+      documentation: `A timer that is used to determine whether waitTime
+        has passed since the last call to addFragment.`,
+      name: 'timer_',
+    },
+    {
+      documentation: `A map containing canonical IDL files. The key of the map
+        corresponds to the name of the definition.`,
+      name: 'canonicalMap_',
+      factory: function() { return {}; },
+    },
+  ],
+
+  methods: [
+    {
+      documentation: 'Takes two ASTs and merges the members from src to dst.',
+      name: 'merge_',
+      code: function(dst, src) {
+        var isDstPartial = dst.definition.isPartial;
+        var isSrcPartial = src.definition.isPartial;
+
+        // Determine if we have multiple non-partial fragments.
+        if (!isDstPartial && !isSrcPartial) {
+          this.error(`Canonicalizer: Two non-partial fragments!
+              In Map: ${foam.json.Pretty.stringify(dst.sources)},
+              Current: ${foam.json.Pretty.stringify(src.sources)}`);
+          throw new Error(`Canonicalizer: Two non-partial fragments were ` +
+              `found for the same definition (${dst.id})!`);
+        }
+
+        // Copy attributes and members from definition into canonical AST.
+        dst.attrs = dst.attrs.concat(src.attrs);
+        dst.definition.members = dst.definition.members
+            .concat(src.definition.members);
+
+        // Copy over inheritance information.
+        if (!dst.definition.inheritsFrom)
```

Comment:
Yes that is correct. I looked at the grammar for WebIDL and it does not seem like non-partials are allowed to inherit